build/cmake/setup.cmake

```diff
@@ -442,6 +442,14 @@ if(UNIX)
         check_symbol_exists(inet_aton arpa/inet.h HAVE_INET_ATON)
         check_symbol_exists(inet_addr arpa/inet.h HAVE_INET_ADDR)
     endif(wxUSE_SOCKETS)
+
+    if(NOT APPLE)
```

Comment:
Can we restrict this to Linux? Checking for `linux/foo` under non-Linux systems doesn't seem to make sense...

Reply:
It is. This is all within `if(UNIX)`.

Reply:
Sorry, but just to be clear, I meant only _Linux_, not any (other) Unix.

Reply:
I don't think CMake has any specific variables for that.
And `unix/joystick.cpp` (that has `#include <linux/joystick.h>` is added inside an `if(UNIX)` check too (via `wx_append_sources(CORE_SRC XWIN_LOWLEVEL)`). But it is also inside `if(WXGTK)`, so I can add that here too.

Reply:
There is `CMAKE_SYSTEM_NAME` which should be `Linux` under Linux, but, well, it's not that important.

Reply:
If we assume it is available on Linux only, then something like this could also work:
```cmake
    if(WXGTK AND NOT APPLE AND NOT CMAKE_SYSTEM_NAME STREQUAL "Linux")
        message(WARNING "wxJoystick is not available")
        wx_option_force_value(wxUSE_JOYSTICK OFF)
    endif()
```

But I find checking if the header exists more robust.

Reply:
It's really not very important, but, as it is right now, you'll be getting this warning message during every cmake configure run under non-Linux systems (e.g. FreeBSD tested by our CI now) and there is nothing you can do to get rid of it, so I thought it would be better to not test for this (Linux-specific) header at all and not warn about its absence neither.

Granted, this also happens when using configure but nobody pays any attention to configure warnings anyhow :-/ and CMake output is a bit tidier, so the warning is more noticeable there.

Reply:
Maybe a better solution is then to disable `wxUSE_JOYSTICK` by default on non-linux systems. And if they enable it manually, they get the warning.